tools/building.py

```diff
@@ -566,6 +566,14 @@ def closure_compiler(filename, advanced=True, extra_closure_args=None):
   if settings.MODULARIZE:
     CLOSURE_EXTERNS += [path_from_root('src/closure-externs/modularize-externs.js')]
 
+  # The ES module exports are generated after closure is run, so add the exports
+  # to and extern file to avoid closure complaining about undefined variables.
+  if settings.MODULARIZE == 'instance':
+    temp = shared.get_temp_files().get('.js', prefix='emcc_closure_externs_').name
+    exports = settings.EXPORTED_FUNCTIONS + settings.EXPORTED_RUNTIME_METHODS
+    utils.write_file(temp, ';\n'.join(['var' +  ' __exp_' + export for export in exports]) + ';\n')
```

Comment:
Can you follow the pattern used below for `emcc_module_exports`?   i.e. `.write(..)` and `.close()`?